toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexService.kt

```diff
@@ -133,21 +155,50 @@ class CratesLocalIndexService : PersistentStateComponent<CratesLocalIndexState>,
         }
     }
 
-    private fun reloadCrates(revTree: RevTree) {
+    fun updateCrates(currentHeadHash: String, prevHeadHash: String) {
+        val reader = repository.newObjectReader()
+        val currentTreeIter = CanonicalTreeParser().apply {
+            val currentHeadTree = repository.resolve("$currentHeadHash^{tree}") ?: return
+            reset(reader, currentHeadTree)
+        }
+
+        val filter = run {
+            val prevHeadTree = repository.resolve("$prevHeadHash^{tree}") ?: return@run TreeFilter.ALL
+
+            val prevTreeIter = CanonicalTreeParser()
+            prevTreeIter.reset(reader, prevHeadTree)
```

Comment:
Consider
```Kotlin
val prevTreeIter = CanonicalTreeParser().apply {
    reset(reader, prevHeadTree)   
}
```
since you're using such construction above